osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderSelectionBlueprint.cs

```diff
@@ -249,6 +251,75 @@ private void removeControlPoints(List<PathControlPoint> toRemove)
             HitObject.Position += first;
         }
 
+        private void splitControlPoints(List<PathControlPoint> toSplit)
+        {
+            // Ensure that there are any points to be split
+            if (toSplit.Count == 0)
+                return;
+
+            foreach (var c in toSplit)
+            {
+                if (c == controlPoints[0] || c == controlPoints[^1] || c.Type is null)
+                    continue;
+
+                // Split off the section of slider before this control point so the remaining control points to split are in the latter part of the slider.
+                var splitControlPoints = controlPoints.TakeWhile(current => current != c).ToList();
+
+                if (splitControlPoints.Count == 0)
+                    continue;
+
+                foreach (var current in splitControlPoints)
+                {
+                    controlPoints.Remove(current);
+                }
+
+                splitControlPoints.Add(c);
+
+                // Turn the control points which were split off into a new slider.
+                var samplePoint = (SampleControlPoint)HitObject.SampleControlPoint.DeepClone();
+                samplePoint.Time = HitObject.StartTime;
+                var difficultyPoint = (DifficultyControlPoint)HitObject.DifficultyControlPoint.DeepClone();
+                difficultyPoint.Time = HitObject.StartTime;
+
+                var newSlider = new Slider
+                {
+                    StartTime = HitObject.StartTime,
+                    Position = HitObject.Position + splitControlPoints[0].Position,
```

Comment:
Slider control points are removed from the start of `HitObject` to create `newSlider`, so at that point `splitControlPoints[0].Position` is not zero. There is nothing that really forces the first control point to be zero.

Reply:
But that would imply that the `Position` of `newSlider` is not the same as the position of the original slider? How can this be if `newSlider` is supposed to be the starting segment?

Reply:
My explanation was not completely right. `splitControlPoints[0].Position` is zero on the first split, but it can split in multiple points at the same time. On the second split, the `HitObject` is missing some of the control points, so `splitControlPoints[0].Position` is not zero, and the `HitObject.Position` has not changed, thus `HitObject.Position + splitControlPoints[0].Position` is the actual starting position of the new slider. It accounts for the `HitObject.Position` not updating in the previous splits.